src/Runner.Listener/MessageListener.cs

```diff
@@ -228,6 +240,24 @@ public async Task<TaskAgentMessage> GetNextMessageAsync(CancellationToken token)
                     // Decrypt the message body if the session is using encryption
                     message = DecryptMessage(message);
 
+                    
+                    if (message != null && message.MessageType == BrokerMigrationMessage.MessageType)
+                    {
+                        Trace.Info("BrokerMigration message received. Polling Broker for messages...");
+
+                        var migrationMessage = JsonUtility.FromString<BrokerMigrationMessage>(message.Body);                        
+                        var brokerServer = HostContext.GetService<IBrokerServer>();
+
+                        await brokerServer.ConnectAsync(migrationMessage.BrokerBaseUrl, _creds);
+                        message = await brokerServer.GetRunnerMessageAsync(token, 
+                                                                        _session.SessionId, 
+                                                                        runnerStatus, 
+                                                                        BuildConstants.RunnerPackage.Version,
+                                                                        VarUtil.OS, 
+                                                                        VarUtil.OSArchitecture, 
+                                                                        _settings.DisableUpdate);
+                    }
+
```

Comment:
I'm admittedly new to the Runner code, but don't we need to save some state so that _all future `get /message`_ requests _also_ get routed to the broker?

Reply:
That's a great question. I actually am intentionally _NOT_ doing that. This is so that we can roll the feature flag forward and back as necessary and the changes get reflected ASAP.

Reply:
Ah, right, so _every_ time the runner polls it'll be redirected to Broker like this.